src/Serializer.php

```diff
@@ -272,4 +277,24 @@ protected function getIncluded()
         $include = $this->request->get($this->expandParam);
         return is_string($include) ? preg_split('/\s*,\s*/', $include, -1, PREG_SPLIT_NO_EMPTY) : [];
     }
+
+    /**
+     * Checks Identification of resource object.
+     * http://jsonapi.org/format/#document-resource-object-identification
+     * @param ResourceInterface $resource
+     */
+    protected function checkIdentification(ResourceInterface $resource)
+    {
+        if (!is_string($resource->getId())) {
```

Comment:
According to these changes you will not have an ability to change it to integer. If we really need this feature then at least it should be optional. So that developers will be able to choose or easily change that behavior.